File: electro-server/src/main/java/com/electro/entity/employee/Employee.java
package com.electro.entity.employee;

import com.electro.entity.BaseEntity;
import com.electro.entity.authentication.User;
import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Accessors(chain = true)
@Entity
@Table(name = "employee")
public class Employee extends BaseEntity {
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false, unique = true)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "office_id", nullable = false)
    @JsonBackReference
    private Office office;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", nullable = false)
    @JsonBackReference
    private Department department;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_type_id", nullable = false)
    @JsonBackReference
    private JobType jobType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_level_id", nullable = false)
    @JsonBackReference
    private JobLevel jobLevel;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_title_id", nullable = false)
    @JsonBackReference
    private JobTitle jobTitle;
}

File: electro-client/src/models/Employee.ts
import BaseResponse from 'models/BaseResponse';
import { UserRequest, UserResponse } from 'models/User';
import { OfficeResponse } from 'models/Office';
import { DepartmentResponse } from 'models/Department';
import { JobTypeResponse } from 'models/JobType';
import { JobLevelResponse } from 'models/JobLevel';
import { JobTitleResponse } from 'models/JobTitle';

export interface EmployeeResponse extends BaseResponse {
  user: UserResponse;
  office: OfficeResponse;
  department: DepartmentResponse;
  jobType: JobTypeResponse;
  jobLevel: JobLevelResponse;
  jobTitle: JobTitleResponse;
}

export interface EmployeeRequest {
  user: UserRequest;
  officeId: number;
  departmentId: number;
  jobTypeId: number;
  jobLevelId: number;
  jobTitleId: number;
}
